manifests/master/bmc_smtp.pp

```diff
@@ -53,20 +57,28 @@
       default   => 'absent',
     }
 
+    ::xinetd::service { $xinetd_svc_name:
+      ensure       => $ensure,
+      service_type => 'UNLISTED',
+      wait         => 'no',
+      user         => 'nobody',
+      groups       => 'no',
+      group        => 'nobody',
+      bind         => $bind_ip,
+      port         => '25',
+      redirect     => 'localhost 25',
+    }
+
   } else {
     # Make sure xinetd service is absent
-    $ensure = 'absent'
-  }
 
-  ::xinetd::service { 'bmc_smtp':
-    ensure       => $ensure,
-    service_type => 'UNLISTED',
-    wait         => 'no',
-    user         => 'nobody',
-    groups       => 'no',
-    group        => 'nobody',
-    bind         => $bind_ip,
-    port         => '25',
-    redirect     => 'localhost 25',
+    # Really ugly way since we can't include ::xinetd::service without
+    # it enforcing xinetd is running
+    exec { 'rm_svc_file_and_restart':
+      path    => ['/bin', '/usr/bin'],
+      command => "rm ${svc_file_path}; systemctl restart xinetd",
```

Comment:
Will this actually stop the xinetd service if there are no service files under `/etc/xinetd.d/`? 

Or will it just restart it without anything listening in xinetd? If that is the case, perhaps there should be another exec that does a `systemctl stop xinetd` if the `/etc/xinetd.d` directory is empty.

Reply:
It will just restart xinetd so that if bmc_smtp gets changes from true > false the live config updates.

I did think about adding something to disable the xinetd service, but was a little more cautious about stopping and disabling a service vs just restarting it.

RHEL systems by default get some junk service files under /etc/xinetd.d that are all disabled but still present. I could do some grepping to make sure everything under these is disabled, but that can be fragile if some file under there gets some funky formatting. I'll check to see if xinetd has an "official" way to check if there are any running services which might be better.

If not, I can make a careful grep to try and make sure it doesn't match things unexpectedly. Probably won't have too many systems running any xinetd services so might not be a big deal either way.

Reply:
That sounds like a good approach. I can’t think of anything else ASD does these days that would require xinetd. But I think in LSST we have an rsync service that runs via xinetd.

Reply:
IRST had a way to see that xinetd was running, but not really serving anything.

Reply:
On mg the internal networks are non-routed so we do some forwarding for nodes w/o public addresses through nodes that do have public addresses using xinetd. We could probably rework this using a squid proxy instead, though. And/or move the mgmt network to a routable space. But as of now we use xinetd.

Reply:
Possibly stop xinetd if something like `lsof -i | grep xinet` returns nothing? 

I tested this on `lsst-web01` & `lsst-adm01` and it seems to work for listing what xinetd is listening for.

Reply:
That sounds like a good approach. I'm thinking this would be better setup in a different puppet module though. Having something that turns off xinetd if there are no services doesn't really fit in the module that sets up and configures xcat. Maybe over in https://github.com/ncsa/puppet-profile_hardening

Reply:
For now I think just fleshing this out here in the ELSE of this is sufficient. As far as I know we have nothing anywhere else in our NCSA shared profiles that includes `::xinetd`.

Reply:
Understood, I've opened a pull request over in https://github.com/ncsa/puppet-profile_hardening/pull/4 to add the logic for stopping and disabling xinetd